src/Types.jl

```diff
@@ -275,6 +275,18 @@ Base.@kwdef mutable struct Project
     compat::Dict{String,String} = Dict{String,String}()# TODO Dict{String, VersionSpec}
 end
 
+Base.@kwdef mutable struct PackageEntry
+    name::Union{String, Nothing} = nothing
+    version::Union{String, Nothing} = nothing
+    path::Union{String, Nothing} = nothing
+    git_tree_sha::Union{SHA1, Nothing} = nothing
+    pinned::Bool = false
+    repo_url::Union{String, Nothing} = nothing
+    repo_rev::Union{String, Nothing} = nothing
+    deps::Dict{String,UUID} = Dict{String,UUID}()
+end
```

Comment:
Open question: should we have an `other::Dict{String,Any}` field here to preserve other TOML entries? Argument in favor is forwards compatibility: if we introduce fields that are meaningful to future versions of Julia/Pkg we would generally want older versions of Pkg to ignore and preserve those fields.